f5_openstack_agent/lbaasv2/drivers/bigip/lbaas_builder.py

```diff
@@ -443,6 +471,13 @@ def _assure_l7rules_created(self, service):
         for l7rule in l7rules:
             if l7rule['provisioning_status'] != plugin_const.PENDING_DELETE:
                 try:
+                    # ignore L7 rule if its policy is really an ESD
+                    l7policy = self.get_l7policy_for_rule(
+                        service['l7policies'], l7rule)
+                    name = l7policy.get('name', None)
```

Comment:
We should log an ERROR here so that the user is aware that the application of rules on top of L7 ESD policies is not a valid request.